src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -150,6 +163,12 @@ func renderResultsJSON(w io.Writer, series []*ts.Series) {
 		vals := s.Values()
 		for i := 0; i < s.Len(); i++ {
 			dp := vals.DatapointAt(i)
+			// Skip points before the query boundary. Ideal place to adjust these would be at the result node but that would make it inefficient
+			// since we would need to create another block just for the sake of restricting the bounds
+			if dp.Timestamp.Before(params.Start) {
```

Comment:
So the series bounds are always same as the bounds of the block its generated from. The problem is that we sometimes have to fetch blocks for a longer query duration so that some functions can look back enough. However, when we return, we want to skip the excess points.

Reply:
I get the motivation, but at the moment it may iterate through a bunch of unnecessary data points; better to calculate the first valid `i` to start from, rather than trying a bunch and discarding. Not really a big deal with a couple of series, but something like 10,000 series where only the last few datapoints in a 2 hour block  are required, we'll be doing a lot of unnecessary looping